src/gsi/apply_scaledepwgts.f90

```diff
@@ -59,53 +59,57 @@ subroutine init_mult_spc_wgts(jcap_in)
   implicit none
 
   integer(i_kind),intent(in   ) :: jcap_in
-  real(r_kind),allocatable      :: totwvlength(:)
 
   integer(i_kind) i,ii,j,k,l,n,kk,nsigend
   integer(i_kind) ig
   real(r_kind) rwv0,rtem1,rtem2
   real (r_kind):: fwgtofwvlen
-  integer(i_kind) :: l_sum_spc_weights
+  real(r_kind) :: totwvlength
+  logical :: l_sum_spc_weights
 
   ! Spectral scale decomposition is differernt between SDL-cross and SDL-nocross
   if( r_ensloccov4scl < tiny_r_kind )then
-     l_sum_spc_weights = 1
+     l_sum_spc_weights = .false.
```

Comment:
@jderber-NOAA  l_sum_spc_weights is supposed to indicate two options: =0, required the summation of their squared values =1;  when =1, required   the summation of the values of themselves =1 .   Will there be any benefits to change to the current false/true form?

Reply:
Looks like my earlier reply was lost.  Probably forgot to hit  the comment button.

For things that only have two options, it is probably better to have true/false with one being the default and one being the option.  When you use true/false only 1 bit is used.  Also, we can avoid the use of "xxxx == 0" (or ==1) in if statements a bit more efficient. May be good to change the name of the true option to be something like "Use_nonsquared_weight" or something like that.

Reply:
@jderber-NOAA  Thanks for the explanation. Though I don't totally agree with this change, it seems more about the "style" . So, I would go ahead to approve this PR. 
Thanks.